### DISCUSSION TITLE
Directus | Option to maintain a default columns in list view of Entity

### DISCUSSION THREAD

**Discussion by ananth3087** [directus/directus#17354] [Archive]
In Directus, while adding data to Entity( example: Template ), We have a list view, in which we can view the list of templates in table view. 
![image](https://user-images.githubusercontent.com/38585821/215857279-439dbc30-e46e-44cd-a90f-46937fd8a261.png)

There is an option to add columns in the list view, but this columns are currently added to table by current user and its saved as current user preference.

![image](https://user-images.githubusercontent.com/38585821/215857307-801f55c1-646c-4138-b170-2e068754ec15.png)

Do we have any option to configure this default columns for all the users from Data model or User based or Role based configuration by Admin user. So that we can maintain the list view of Table to have same columns for all users example Template ID, Template Name, Short Description, Status.
![image](https://user-images.githubusercontent.com/38585821/215857337-720fb199-9ac1-47b2-aab0-6f1489f503af.png)

**Comment by rasco** [comment]
Yes, I see this is an important usability feature.

It's "kind of possible", but it's a bit complicated.
Basically you have to login delete all users templates/bookmarks and then configure the "default"/"admin" templates.

**Comment by ImaCrea** [comment]
Hey, I found out that if you set a favorite and set it with a collection, with global view and **no name** then it seems to overwrite the default view of the collection. That's how I did it.

![Capture d’écran 2023-04-04 à 17 53 35](https://user-images.githubusercontent.com/764332/229848825-d9223436-78b4-49d2-8985-271cd9ae97ee.png)

Note : I set that before inviting users. If after, you might have to delete some user-level favorite of that type as suggest @rasco

**Reply by ImaCrea** [reply]
correction : not so sure it works as I see when user connects back to it a new view is created for this user. At least it keeps the global one set before but if they change then bye bye "default" favorite.

which is understandable, it's important users may not be forced with a default configuration. but if as an admin you're required to update the "default" view for everyone, then you'll have to delete all these favorites at user level and carefully not delete personal ones they'd want to keep..

not ideal at all. There should be a "default" view you could set in the settings of any collection imho.

**Reply by rijkvanzanten** [reply]
The way it works is that it is "layered". You can set a global default and/or a default per role, and whatever changes the user makes on top of that are saved for just that user 👍🏻 

If it wouldn't save the users preferences, any change they make on the layout (including little things like sort, column sizes, visible fields, card size etc etc) will be lost every time they navigate away

**Reply by rasco** [reply]
@rijkvanzanten but the user has no option to revert back to default.
and yes, the users are too dumb. they **need** a stable default layout that is always available.

also, as @ImaCrea said, it's quite troublesome to reset all users' layouts for tables to the default one. 
you have to be careful to select the right table, and the right layouts, so you don't delete the default one or the wrong tables.

**Reply by ImaCrea** [reply]
> @rijkvanzanten but the user has no option to revert back to default. and yes, the users are too dumb. they **need** a stable default layout that is always available.
> 
> also, as @ImaCrea said, it's quite troublesome to reset all users' layouts for tables to the default one. you have to be careful to select the right table, and the right layouts, so you don't delete the default one or the wrong tables.

I agree with the idea of letting user kind of "reset" the table view if they'd like. Though, I'd never say it's because they're "dumb", all of us need some simplicity from time to time ;)

**Comment by rasco** [comment]
Something just happened to me, that has also happened to one of our users before, and is related to this:

I had an error on the collection page and it showed a button to "reset my layout" or something (I don't remember the exact wording).

However, after pressing that button what I ended up with was this:

<img width="1298" alt="Screenshot 2023-05-10 at 12 16 36 PM" src="https://github.com/directus/directus/assets/289536/31b55e08-1d2f-46a8-9d2b-b79c53011f1f">

A regular user would be completely dumbfounded as to what just happened, and how to fix it.

So just adding a button like "reset layout" or something would be very important for cases like this.
